website/docs/plugin/framework/resources/plan-modification.mdx

```diff
@@ -62,45 +66,43 @@ The framework implements some common use case modifiers in the typed packages un
 
 ```go
-// stringDefaultModifier is a plan modifier that sets a default value for a
-// types.StringType attribute when it is not configured. The attribute must be
-// marked as Optional and Computed. When setting the state during the resource
-// Create, Read, or Update methods, this default value must also be included or
-// the Terraform CLI will generate an error.
-type stringDefaultModifier struct {
-    Default string
+// timeNowDefaultModifier is a plan modifier that sets the value for a
+// types.StringType attribute to the current time when it is not configured.
+// The attribute must be marked as Optional and Computed. When setting the
+// state during the resource Create, Read, or Update methods, this value
+// must also be included or the Terraform CLI will generate an error.
+type timeNowDefaultModifier struct {
 }
 
-// Description returns a plain text description of the validator's behavior, suitable for a practitioner to understand its impact.
-func (m stringDefaultModifier) Description(ctx context.Context) string {
-    return fmt.Sprintf("If value is not configured, defaults to %s", m.Default)
+// Description returns a plain text description of the plan modifier's behavior, suitable for a practitioner to understand its impact.
+func (m timeNowDefaultModifier) Description(ctx context.Context) string {
+    return fmt.Sprintf("If value is not configured, defaults to a string representation of the current time")
 }
 
-// MarkdownDescription returns a markdown formatted description of the validator's behavior, suitable for a practitioner to understand its impact.
-func (m stringDefaultModifier) MarkdownDescription(ctx context.Context) string {
-    return fmt.Sprintf("If value is not configured, defaults to `%s`", m.Default)
+// MarkdownDescription returns a markdown formatted description of the plan modifier's behavior, suitable for a practitioner to understand its impact.
+func (m timeNowDefaultModifier) MarkdownDescription(ctx context.Context) string {
+    return fmt.Sprintf("If value is not configured, defaults to a string representation of the current time")
 }
 
 // PlanModifyString runs the logic of the plan modifier.
 // Access to the configuration, plan, and state is available in `req`, while
 // `resp` contains fields for updating the planned value, triggering resource
 // replacement, and returning diagnostics.
-func (m stringDefaultModifier) PlanModifyString(ctx context.Context, req planmodifier.StringRequest, resp *planmodifier.StringResponse) {
+func (m timeNowDefaultModifier) PlanModifyString(ctx context.Context, req planmodifier.StringRequest, resp *planmodifier.StringResponse) {
     // If the value is unknown or known, do not set default value.
     if !req.PlanValue.IsNull() {
         return
     }
 
-    resp.PlanValue = types.StringValue(m.Default)
+    resp.PlanValue = types.StringValue(time.Now().String())
```

Comment:
How do you feel about adjusting this to be a plan modifier that normalizes JSON-encoded strings (keeping the prior state if they match when normalized) so its separate from the default value use case?

Reply:
As discussed, have used one of the predefined `UseStateForUnknown` plan modifiers as an example.